test/CSBondCurve.t.sol

```diff
@@ -379,4 +440,327 @@ contract CSBondCurveTest is Test {
+                        intervals[i + 1].trend;
+                    if (bondToCheck > maxBondInInterval) {
+                        bondToCheck = maxBondInInterval;
+                    }
+                    neededIndex = i;
+                    break;
+                }
+            }
+            neededIndex = i;
+        }
+
+        return
+            intervals[neededIndex].fromKeysCount +
+            (bondToCheck - fromBondAcc) /
+            intervals[neededIndex].trend;
+    }
+
+    function prepareInputs(
+        uint256[] memory fromKeysCount,
+        uint256[] memory trend,
+        uint256 keysToCheck,
+        uint256 bondToCheck
+    ) public pure returns (uint256[2][] memory, uint256, uint256) {
+        vm.assume(fromKeysCount.length > 0);
+        vm.assume(trend.length > 0);
+
+        // Assume: intervals.length > 0
+        uint256 intervalsCount = Math.max(
+            1,
+            Math.min(fromKeysCount.length, trend.length) %
+                MAX_BOND_CURVE_INTERVALS_COUNT
+        );
+        for (uint256 i = 0; i < intervalsCount; ++i) {
+            // Assume: fromKeysCount[i] > 0
+            fromKeysCount[i] = Math.max(
+                1,
+                fromKeysCount[i] % MAX_FROM_KEYS_COUNT_VALUE
+            );
+            // Assume: trend[i] > 0
+            trend[i] = Math.max(1 wei, trend[i] % MAX_TREND_VALUE);
+        }
+        assembly ("memory-safe") {
+            // Shrink `fromKeysCount` and `trend` arrays to `intervalsCount`
+            mstore(fromKeysCount, intervalsCount)
+            mstore(trend, intervalsCount)
+        }
+
+        // Assume: fromKeysCount[i] < fromKeysCount[i + 1]
+        uint256 n = fromKeysCount.length;
+        for (uint256 i = 0; i < n; i++) {
+            for (uint256 j = 0; j < n - 1; j++) {
+                if (fromKeysCount[j] > fromKeysCount[j + 1]) {
+                    (fromKeysCount[j], fromKeysCount[j + 1]) = (
+                        fromKeysCount[j + 1],
+                        fromKeysCount[j]
+                    );
+                }
+                if (fromKeysCount[j] == fromKeysCount[j + 1]) {
+                    // Make it different because we need to have unique values
+                    fromKeysCount[j + 1] = fromKeysCount[j] + 1;
+                }
+            }
+        }
+        // Assume: first interval starts from "1" keys count
+        fromKeysCount[0] = 1;
+
+        assertEq(fromKeysCount.length, trend.length);
+
+        // Dev: zip `fromKeysCount` and `trend` arrays to `uint256[2][] intervals`
+        uint256[2][] memory _bondCurve = new uint256[2][](fromKeysCount.length);
+        for (uint256 i = 0; i < intervalsCount; ++i) {
+            _bondCurve[i] = [fromKeysCount[i], trend[i]];
+        }
+        keysToCheck = bound(keysToCheck, 1, MAX_FROM_KEYS_COUNT_VALUE);
+        bondToCheck = bound(
+            bondToCheck,
+            trend[0],
+            MAX_TREND_VALUE * MAX_FROM_KEYS_COUNT_VALUE
```

Comment:
Why not to check very large values, i.e. definitely out of range of the last interval?

Reply:
It can be just `type(uint256).max`